package.json

```diff
@@ -49,11 +49,12 @@
     "check": "yarn lint:check && yarn prettier:check",
     "fix": "yarn lint:fix && yarn prettier:fix",
     "clean": "rm -rf lib/* deno/lib/*",
-    "build": "yarn run clean && npm run build:cjs && npm run build:esm && npm run build:deno",
+    "build": "yarn run clean && npm run build:cjs && npm run build:esm && npm run build:deno && yarn run build:tweak-dts",
     "build:deno": "node ./deno/build.mjs",
     "build:esm": "rollup --config rollup.config.js",
     "build:cjs": "tsc --p tsconfig.cjs.json",
     "build:types": "tsc --p tsconfig.types.json",
+    "build:tweak-dts": "echo 'export as namespace Zod;' >> lib/index.d.ts",
```

Comment:
Is there something we can do in the source to get this same effect without having to patch the artifact?

Reply:
I find way to do it without patch to generated dts! can you review it?

Reply:
Definitely better! Can you give me a little context as to why we need to export a namespace for the UMD target? Is that because it adds a global binding to the library?

Reply:
> Is that because it adds a global binding to the library?

Yes!

FYI: If project uses module (like `import ...`), global binding will be disabled (tsc says `'Zod' refers to a UMD global, but the current file is a module. Consider adding an import instead.ts(2686)`)